src/Ledger/GovernanceActions.lagda

```diff
@@ -54,14 +54,27 @@ record Anchor : Type where
     url   : String
     hash  : DocHash
 
-data GovAction : Type where
-  NoConfidence     :                                             GovAction
-  UpdateCommittee  : (Credential ⇀ Epoch) → ℙ Credential → ℚ  →  GovAction
-  NewConstitution  : DocHash → Maybe ScriptHash               →  GovAction
-  TriggerHF        : ProtVer                                  →  GovAction
-  ChangePParams    : PParamsUpdate                            →  GovAction
-  TreasuryWdrl     : (RwdAddr ⇀ Coin)                         →  GovAction
-  Info             :                                             GovAction
+data GovActionType : Type where
+  NoConfidence     : GovActionType
+  UpdateCommittee  : GovActionType
+  NewConstitution  : GovActionType
+  TriggerHF        : GovActionType
+  ChangePParams    : GovActionType
+  TreasuryWdrl     : GovActionType
+  Info             : GovActionType
+
+GovActionData : GovActionType → Type
+GovActionData = λ where
```

Comment:
I think this lambda is a bit too unusual in this place. People are probably going to understand it, but they might be confused why it wasn't written in the "normal" way like everything else. I don't think it's too repetitive to match on the LHS here.